Improve error when not authenticated

The extension uses application-default credentials. When not present or expired, the experience is not very clear.

<img width="421" alt="Image" src="https://github.com/user-attachments/assets/2af3382b-d64a-47ee-a9f1-f9ee58a87db8" />

Here's what goes in the logs:

```
21:23:53.667	[JavaFX Application Thread]	ERROR	dimilab.qupath.ext.omezarr.CloudOmeZarrServerBuilder	Cloud OME-Zarr couldn't open gs://spatialomics-data-us-central1/cellular-segmentation/ome/perkin-elmer/LuCa-7color/output-1024x1024.zarr: Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object. Permission 'storage.objects.get' denied on resource (or it may not exist).	com.google.cloud.storage.StorageException: Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object. Permission 'storage.objects.get' denied on resource (or it may not exist).
	at com.google.cloud.storage.StorageException.translate(StorageException.java:179)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.translate(HttpStorageRpc.java:330)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.get(HttpStorageRpc.java:558)
	at com.google.cloud.storage.StorageImpl.lambda$internalGetBlob$65(StorageImpl.java:1627)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:103)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
	at com.google.cloud.storage.Retrying.run(Retrying.java:65)
	at com.google.cloud.storage.StorageImpl.run(StorageImpl.java:1557)
	at com.google.cloud.storage.StorageImpl.internalGetBlob(StorageImpl.java:1625)
	at com.google.cloud.storage.StorageImpl.get(StorageImpl.java:342)
	at com.google.cloud.storage.StorageImpl.get(StorageImpl.java:347)
	at com.google.cloud.storage.contrib.nio.CloudStorageFileSystemProvider.readAttributes(CloudStorageFileSystemProvider.java:825)
	at java.base/java.nio.file.spi.FileSystemProvider.readAttributesIfExists(Unknown Source)
	at java.base/java.nio.file.Files.isDirectory(Unknown Source)
	at com.bc.zarr.ZarrUtils.ensureDirectory(ZarrUtils.java:157)
	at com.bc.zarr.ZarrGroup.open(ZarrGroup.java:95)
	at dimilab.qupath.ext.omezarr.CloudOmeZarrServer.<init>(CloudOmeZarrServer.kt:69)
	at dimilab.qupath.ext.omezarr.CloudOmeZarrServerBuilder.buildServer(CloudOmeZarrServerBuilder.kt:84)
	at qupath.lib.images.servers.ImageServerBuilder$DefaultImageServerBuilder.buildOriginal(ImageServerBuilder.java:350)
	at qupath.lib.images.servers.ImageServerBuilder$AbstractServerBuilder.build(ImageServerBuilder.java:174)
	at qupath.lib.projects.DefaultProject$DefaultProjectImageEntry.readImageData(DefaultProject.java:693)
	at qupath.lib.gui.QuPathGUI.openImageEntry(QuPathGUI.java:1486)
	at qupath.lib.gui.panes.ProjectBrowser.lambda$new$5(ProjectBrowser.java:223)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:232)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:189)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        ...
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$ClickGenerator.postProcess(Scene.java:3684)
	at javafx.scene.Scene$MouseHandler.process(Scene.java:3989)
	at javafx.scene.Scene.processMouseEvent(Scene.java:1890)
	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2704)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:411)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:301)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:450)
	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:449)
	at com.sun.glass.ui.View.handleMouseEvent(View.java:551)
	at com.sun.glass.ui.View.notifyMouse(View.java:937)
	at com.sun.glass.ui.mac.MacView.notifyMouse(MacView.java:127)
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 401 Unauthorized
GET https://storage.googleapis.com/storage/v1/b/spatialomics-data-us-central1/o/cellular-segmentation%2Fome%2Fperkin-elmer%2FLuCa-7color%2Foutput-1024x1024.zarr?projection=full
{
  "code" : 401,
  "errors" : [ {
    "domain" : "global",
    "location" : "Authorization",
    "locationType" : "header",
    "message" : "Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object. Permission 'storage.objects.get' denied on resource (or it may not exist).",
    "reason" : "required"
  } ],
  "message" : "Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object. Permission 'storage.objects.get' denied on resource (or it may not exist)."
}
	at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:146)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:118)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:37)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$3.interceptResponse(AbstractGoogleClientRequest.java:479)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1111)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:565)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:506)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:616)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.get(HttpStorageRpc.java:555)
	... 61 more
```

Then the exception which is almost (but not quite?) displayed in the error message:

```
21:23:53.700	[JavaFX Application Thread]	ERROR	qupath.lib.gui.QuPathGUI	Unable to build ImageServer for gs://spatialomics-data-us-central1/cellular-segmentation/ome/perkin-elmer/LuCa-7color/output-1024x1024.zarr with requested provider dimilab.qupath.ext.omezarr.CloudOmeZarrServerBuilder	java.io.IOException: Unable to build ImageServer for gs://spatialomics-data-us-central1/cellular-segmentation/ome/perkin-elmer/LuCa-7color/output-1024x1024.zarr with requested provider dimilab.qupath.ext.omezarr.CloudOmeZarrServerBuilder
	at qupath.lib.images.servers.ImageServerBuilder$DefaultImageServerBuilder.buildOriginal(ImageServerBuilder.java:369)
	at qupath.lib.images.servers.ImageServerBuilder$AbstractServerBuilder.build(ImageServerBuilder.java:174)
	at qupath.lib.projects.DefaultProject$DefaultProjectImageEntry.readImageData(DefaultProject.java:693)
	at qupath.lib.gui.QuPathGUI.openImageEntry(QuPathGUI.java:1486)
	at qupath.lib.gui.panes.ProjectBrowser.lambda$new$5(ProjectBrowser.java:223)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:232)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:189)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
        ...
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$ClickGenerator.postProcess(Scene.java:3684)
	at javafx.scene.Scene$MouseHandler.process(Scene.java:3989)
	at javafx.scene.Scene.processMouseEvent(Scene.java:1890)
	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2704)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:411)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:301)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:450)
	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:449)
	at com.sun.glass.ui.View.handleMouseEvent(View.java:551)
	at com.sun.glass.ui.View.notifyMouse(View.java:937)
	at com.sun.glass.ui.mac.MacView.notifyMouse(MacView.java:127)
```

One way or another this should be improved. Perhaps by raising an alert dialog? 🤔 
At minimum we should output an explicit log.

--- a/src/main/kotlin/dimilab/qupath/ext/omezarr/CloudOmeZarrServerBuilder.kt
+++ b/src/main/kotlin/dimilab/qupath/ext/omezarr/CloudOmeZarrServerBuilder.kt
@@ -1,16 +1,23 @@
 package dimilab.qupath.ext.omezarr
 
+import com.google.api.client.googleapis.json.GoogleJsonResponseException
+import com.google.cloud.storage.StorageException
 import qupath.lib.images.servers.ImageServer
 import qupath.lib.images.servers.ImageServerBuilder
 import qupath.lib.images.servers.ImageServerBuilder.DefaultImageServerBuilder
 import qupath.lib.images.servers.ImageServerBuilder.UriImageSupport
 import qupath.lib.images.servers.ImageServerMetadata
 import java.awt.image.BufferedImage
+import java.io.IOException
 import java.net.URI
 
 class CloudOmeZarrServerBuilder : ImageServerBuilder<BufferedImage> {
   companion object {
     private val logger = org.slf4j.LoggerFactory.getLogger(CloudOmeZarrServerBuilder::class.java)
+  }
+
+  class UnauthorizedError : IOException {
+    constructor(message: String) : super(message)
   }
 
   override fun checkImageSupport(uri: URI?, vararg args: String?): UriImageSupport<BufferedImage>? {
@@ -82,6 +89,16 @@ class CloudOmeZarrServerBuilder : ImageServerBuilder<BufferedImage> {
 
     try {
       return CloudOmeZarrServer(uri, *args)
+    } catch (e: StorageException) {
+      if (e.cause is GoogleJsonResponseException) {
+        val responseException = e.cause as GoogleJsonResponseException
+        if (responseException.statusCode == 401) {
+          logger.error("Error: not authorized to access {}: {}", uri, responseException.message)
+          throw UnauthorizedError("Access error. Quit QuPath and run this: gcloud auth application-default login")
+        }
+      }
+      logger.error("Google Cloud Storage exception: {}", e.message, e)
+      return null
     } catch (e: Exception) {
       logger.error("Cloud OME-Zarr couldn't open {}: {}", uri, e.message, e)
       return null

